File: #26.Remove Duplicates from Sorted Array/Remove Duplicates from Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {

    if(nums.length==0){
        return 0;
    }
    if(nums.length==1){
        return 1;
    }

    var pos=1,i=1;
    while(i<nums.length){
        if(nums[i]!= nums[pos-1]){
            nums[pos]=nums[i];
            ++i;
            ++pos;
        }else{
            ++i;
        }
    }
    return pos;
};

File: #26.Remove Duplicates from Sorted Array/Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size()==0) return 0;
        if(nums.size()==1) return 1;
        int point=1;
        int i=1;
        while(i<nums.size()){
            if(nums[i]!=nums[point-1]){
                nums[point]=nums[i];
                ++i;
                ++point;
            }else{
                ++i;
            }
        }
        return point;
    }
};
